Build failure: `mitscheme`

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `mitscheme` from latest master

### Build log

```
[...]
mkdir /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
/nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install -c --preserve-timestamps -m 644 mit-scheme-ffi/* /nix/store/yl0rnsdx4ypaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi/.
/nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install: cannot stat 'mit-scheme-ffi/*': No such file or directory
make[1]: *** [../make-common:159: install-html] Error 1
make[1]: Leaving directory '/build/mit-scheme-11.2/doc/ffi'
make: *** [Makefile:129: install-html] Error 1
make: Leaving directory '/build/mit-scheme-11.2/doc'
error: builder for '/nix/store/f0jidkm5wr8r9b03k7bpbx2840x1h0i9-mit-scheme-11.2.drv' failed with exit code 2;
       last 10 log lines:
       > texi2any --html --no-split ffi.texinfo
       > rm -rf /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > /nix/store/j80wcbm6bb1nsqsgmniq43rg980p27q9-bash-5.2-p15/bin/bash ../mkinstalldirs /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-cheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > mkdir /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > /nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install -c --preserve-timestamps -m 644 mit-scheme-ffi/* /nix/store/y0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi/.
       > /nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install: cannot stat 'mit-scheme-ffi/*': No such file or directory
       > make[1]: *** [../make-common:159: install-html] Error 1
       > make[1]: Leaving directory '/build/mit-scheme-11.2/doc/ffi'
       > make: *** [Makefile:129: install-html] Error 1
       > make: Leaving directory '/build/mit-scheme-11.2/doc'
       For full logs, run 'nix log /nix/store/f0jidkm5wr8r9b03k7bpbx2840x1h0i9-mit-scheme-11.2.drv'.
```

### Additional context

I have not seen this build successfully on my machine since before March 18. I imagine there was some wide-reaching change merged in from `staging-next` that caused this.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.26, NixOS, 23.05 (Stoat), 23.05.20230501.8b3bc69`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15740,8 +15740,10 @@ with pkgs;
 
   mitama-cpp-result = callPackage ../development/libraries/mitama-cpp-result { };
 
-  mitscheme = callPackage ../development/compilers/mit-scheme
-    { texLive = texlive.combine { inherit (texlive) scheme-small epsf texinfo; }; };
+  mitscheme = callPackage ../development/compilers/mit-scheme {
+    texLive = texlive.combine { inherit (texlive) scheme-small epsf texinfo; };
+    texinfo = texinfo6;
+  };
 
   mitschemeX11 = mitscheme.override {
     enableX11 = true;

